### ISSUE TITLE
Code completion is not full for custom functions

### ISSUE THREAD

**Issue by smedelyan** [godotengine/godot#27467]
**Godot version:** 3.1

**OS/device including version:** Ubuntu 18.04

**Issue description:**
Code completion does not provide parameter hints for custom functions if they are used from a script other than they are declared in.

```gdscript
# one script
extends Object
class_name KnownClass

func test(param: String):
	pass
```

```gdscript
# another script
extends Node2D

func _ready():
	var kc: KnownClass = KnownClass.new()
	kc.test("here completion does not show parameter list")
```

**Minimal reproduction project:**
[repro_cc_custom_func.tar.gz](https://github.com/godotengine/godot/files/3015179/repro_cc_custom_func.tar.gz)

**Comment by Ranoller** [comment]
You are right, i can confirm that.... but from long ago. And it´s not consistent, some times code completion doesn´t provide parameters of script self functions once "(" character is reached. And sometimes doesn´t provide autocompletion for self script functions at all.

**Comment by smedelyan** [comment]
@Ranoller then the time has come to finally solve this annoying one! =)

**Comment by HolonProduction** [comment]
Does not apply to 4.x

**Comment by lawnjelly** [comment]
Closed by #68449 .
